climada/hazard/tc_tracks_synth.py

```diff
@@ -726,37 +748,60 @@ def _apply_decay_coeffs(track, v_rel, p_rel, land_geom, s_rel):
             continue
         if land_sea - sea_land == 1:
             continue
-        p_decay = _calc_decay_ps_value(track, p_landfall, land_sea - 1, s_rel)
-        p_decay = _decay_p_function(p_decay, p_rel[ss_scale_idx][1],
-                                    track.dist_since_lf[sea_land:land_sea].values)
-        # dont applay decay if it would decrease central pressure
-        p_decay[p_decay < 1] = track.central_pressure[sea_land:land_sea][p_decay < 1] / p_landfall
-        track.central_pressure[sea_land:land_sea] = p_landfall * p_decay
+        S = _calc_decay_ps_value(track, p_landfall, land_sea - 1, s_rel)
+        if S <= 1:
+            # central_pressure at start of landfall > env_pres after landfall:
+            # set central_pressure to environmental pressure during whole lf
+            track.central_pressure[sea_land:land_sea] = track.environmental_pressure[sea_land:land_sea]
+        else:
+            p_decay = _decay_p_function(S, p_rel[ss_scale_idx][1],
+                                        track.dist_since_lf[sea_land:land_sea].values)
+            # dont applay decay if it would decrease central pressure
+            if np.any(p_decay > 1):
+                LOGGER.warning('Landfall decay would decrease pressure for '
```

Comment:
If despite the check on `S` the decay would increase pressure drop (decrease pressure), it is not applied (as was before) but the user is also warned that this is the case.

Reply:
Should this be a warning or an error? What does 'Decay not applied, please check' mean? How should a user know what to do in this situation?

Reply:
This actually cannot happen, because we checked that `S > 1` and `p_rel > 0`, but I still put that here in case future code changes so `p_rel` or `S` calculations would make it again possible for `p_decay` to be `>1`.

Reply:
Not sure if I like this, but then one should probably raise an error right? Or is it fine if the decay is not applied? Also, please update the message to something more useful. 'Please check' does not give me any actual information about the issue.